sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/annotation/ForbidAnnotationElementValueCheck.java

```diff
@@ -0,0 +1,386 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2016 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.sevntu.checkstyle.checks.annotation;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.google.common.base.CharMatcher;
+import com.puppycrawl.tools.checkstyle.api.Check;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * <p>
+ * Forbids specific
+ * <a href= 'https://docs.oracle.com/javase/specs/jls/se7/html/jls-9.html#jls-9.6.1'>element
+ * value</a> for specific annotation. You can configure this check using following options:
+ * </p>
+ * <ul>
+ * <li>Annotation name</li>
+ * <li>Annotation element name</li>
+ * <li>Forbidden annotation element value pattern</li>
+ * </ul>
+ * Example of usage:<br>
+ * <p>
+ * Here is XML configs and according code samples needed to forbid
+ * </p>
+ * <p>
+ * Config
+ * </p>
+ *
+ * <pre>
+ * &lt;module name="ForbidAnnotationElementValue"&gt;
+ * &lt;property name="annotationName" value="Bean"/&gt;
+ * &lt;property name="elementName" value="name"/&gt;
+ * &lt;property name="forbiddenElementValueRegexp" value="AnnotationConfigUtils\..*"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Code
+ * </p>
+ *
+ * <pre>
+ *  &#64;Bean(name = AnnotationConfigUtils.SCHEDULED_ANNOTATION_PROCESSOR_BEAN_NAME)

```

Comment:
lets be more practical in examples and do real users problems:
"usage of "expected" parameter in annotation Test"
in web we could find smth else.